src/storagesync/azext_storagesync/tests/latest/test_storagesync_scenario.py

```diff
@@ -0,0 +1,189 @@
+                         JMESPathCheck('syncGroupStatus', 0)])
+
+        self.cmd('az storagesync sync-group show '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name} '
+                 '--name {sync_group_name}',
+                 checks=[JMESPathCheck('name', sync_group_name),
+                         JMESPathCheck('syncGroupStatus', 0)])
+
+        self.cmd('az storagesync sync-group list '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name}',
+                 checks=[JMESPathCheck('length(@)', 1)])
+
+        self.cmd('az storagesync cloud-endpoint create '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name} '
+                 '--sync-group-name {sync_group_name} '
+                 '--name {cloud_endpoint_name} '
+                 '--storage-account {sa} '
+                 '--azure-file-share-name {file_share_name}',
+                 checks=[JMESPathCheck('provisioningState', 'Succeeded')])
+
+        self.cmd('az storagesync cloud-endpoint show '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name} '
+                 '--sync-group-name {sync_group_name} '
+                 '--name {cloud_endpoint_name}',
+                 checks=[JMESPathCheck('name', cloud_endpoint_name)])
+
+        self.cmd('az storagesync cloud-endpoint list '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name} '
+                 '--sync-group-name {sync_group_name}',
+                 checks=[JMESPathCheck('length(@)', 1)])
+
+        server_list = self.cmd('az storagesync registered-server list '
+                               '--resource-group {rg} '
+                               '--storage-sync-service-name {sync_service_name}').get_output_in_json()
+        self.assertEqual(1, len(server_list))
+
+        server_id = server_list[0]['serverId']
+        self.kwargs.update({
+            'server_id': server_id
+        })
+
+        self.cmd('az storagesync registered-server show '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name} '
+                 '--server-id {server_id}',
+                 checks=[JMESPathCheck('provisioningState', 'Succeeded'),
+                         JMESPathCheck('serverId', server_id)])
+
+        self.cmd('az storagesync server-endpoint create '
+                 '--resource-group {rg} '
+                 '--storage-sync-service-name {sync_service_name} '
+                 '--sync-group-name {sync_group_name} '
+                 '--name {server_endpoint_name} '
+                 '--server-id {server_id} '
+                 '--server-local-path "c:\\users\\syncfolder"',
+                 checks=[JMESPathCheck('provisioningState', 'Succeeded'),
```

Comment:
this command can accept four arguments. Could we add more tests? Looks like this should be an important command.

Reply:
I think checking 'provisioningState' is enough, and other properties from response is not so valuable for check. If state is succeeded, it means creation is succeeded.

---
In reply to: [392873316](https://github.com/Azure/azure-cli-extensions/pull/1389#discussion_r392873316) [](ancestors = 392873316)